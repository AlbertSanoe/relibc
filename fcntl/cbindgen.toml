include_guard = "_FCNTL_H"
language = "C"

[enum]
prefix_with_name = true
