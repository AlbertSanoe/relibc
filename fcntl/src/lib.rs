//! fcntl implementation for Redox, following http://pubs.opengroup.org/onlinepubs/7908799/xsh/fcntl.h.html

#![no_std]

extern crate libc;

use libc::*;

pub const F_DUPFD: c_int = 0;
pub const F_GETFD: c_int = 1;
pub const F_SETFD: c_int = 2;
pub const F_GETFL: c_int = 3;
pub const F_SETFL: c_int = 4;
pub const F_GETLK: c_int = 5;
pub const F_SETLK: c_int = 6;
pub const F_SETLKW: c_int = 7;

pub const FD_CLOEXEC: c_int = 0x0100_0000;

pub const F_RDLCK: c_int = 0;
pub const F_WRLCK: c_int = 1;
pub const F_UNLCK: c_int = 2;

pub const O_RDONLY: c_int =     0x0001_0000;
pub const O_WRONLY: c_int =     0x0002_0000;
pub const O_RDWR: c_int =       0x0003_0000;
pub const O_NONBLOCK: c_int =   0x0004_0000;
pub const O_APPEND: c_int =     0x0008_0000;
pub const O_SHLOCK: c_int =     0x0010_0000;
pub const O_EXLOCK: c_int =     0x0020_0000;
pub const O_ASYNC: c_int =      0x0040_0000;
pub const O_FSYNC: c_int =      0x0080_0000;
pub const O_CLOEXEC: c_int =    0x0100_0000;
pub const O_CREAT: c_int =      0x0200_0000;
pub const O_TRUNC: c_int =      0x0400_0000;
pub const O_EXCL: c_int =       0x0800_0000;
pub const O_DIRECTORY: c_int =  0x1000_0000;
pub const O_STAT: c_int =       0x2000_0000;
pub const O_SYMLINK: c_int =    0x4000_0000;
pub const O_NOFOLLOW: c_int =   0x8000_0000;
pub const O_ACCMODE: c_int = O_RDONLY | O_WRONLY | O_RDWR;

#[no_mangle]
pub extern "C" fn creat(path: *const c_char, mode: mode_t) -> c_int {
    open(path, O_WRONLY | O_CREAT | O_TRUNC, mode)
}

#[no_mangle]
pub extern "C" fn fcntl(fildes: c_int, cmd: c_int, arg: c_int) -> c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn open(path: *const c_char, oflag: c_int, mode: mode_t) -> c_int {
    unimplemented!();
}

/*
#[no_mangle]
pub extern "C" fn func(args) -> c_int {
    unimplemented!();
}
*/
