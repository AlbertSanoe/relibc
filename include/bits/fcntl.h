#ifndef _BITS_FCNTL_H
#define _BITS_FCNTL_H

int open(const char* filename, int flags, ...);
int fcntl(int fildes, int cmd, ...);

#endif
