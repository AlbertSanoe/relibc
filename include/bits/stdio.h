#ifndef _BITS_STDIO_H
#define _BITS_STDIO_H

#define EOF (-1)

int fprintf(FILE * stream, const char * fmt, ...);
int printf(const char * fmt, ...);
int snprintf(char *s, size_t n, const char * fmt, ...);
int sprintf(char *s, const char * fmt, ...);

#endif /* _BITS_STDIO_H */
