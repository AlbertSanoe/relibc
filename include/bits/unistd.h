#ifndef _BITS_UNISTD_H
#define _BITS_UNISTD_H

int execl(const char *path, const char* argv0, ...);
int execle(const char *path, const char* argv0, ...);

#endif
