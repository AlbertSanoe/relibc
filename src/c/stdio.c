#include <stdarg.h>
#include <stddef.h>

typedef struct FILE FILE;

int vfprintf(FILE * stream, const char * fmt, va_list ap);

int fprintf(FILE * stream, const char * fmt, ...) {
    int ret;
    va_list ap;
    va_start(ap, fmt);
    ret = vfprintf(stream, fmt, ap);
    va_end(ap);
    return ret;
}

int vprintf(const char * fmt, va_list ap);

int printf(const char * fmt, ...) {
    int ret;
    va_list ap;
    va_start(ap, fmt);
    ret = vprintf(fmt, ap);
    va_end(ap);
    return ret;
}

int vsnprintf(char * s, size_t n, const char * fmt, va_list ap);

int snprintf(char * s, size_t n, const char * fmt, ...) {
    int ret;
    va_list ap;
    va_start(ap, fmt);
    ret = vsnprintf(s, n, fmt, ap);
    va_end(ap);
    return ret;
}

int vsprintf(char * s, const char * fmt, va_list ap);

int sprintf(char *s, const char * fmt, ...) {
    int ret;
    va_list ap;
    va_start(ap, fmt);
    ret = vsprintf(s, fmt, ap);
    va_end(ap);
    return ret;
}
