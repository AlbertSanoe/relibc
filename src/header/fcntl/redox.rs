use platform::types::*;

pub const O_RDONLY: c_int = 0x0001;
pub const O_WRONLY: c_int = 0x0002;
pub const O_RDWR: c_int = 0x0003;
pub const O_NONBLOCK: c_int = 0x0004;
pub const O_APPEND: c_int = 0x0008;
pub const O_SHLOCK: c_int = 0x0010;
pub const O_EXLOCK: c_int = 0x0020;
pub const O_ASYNC: c_int = 0x0040;
pub const O_FSYNC: c_int = 0x0080;
pub const O_CLOEXEC: c_int = 0x0100;
pub const O_CREAT: c_int = 0x0200;
pub const O_TRUNC: c_int = 0x0400;
pub const O_EXCL: c_int = 0x0800;
pub const O_DIRECTORY: c_int = 0x1000;
pub const O_PATH: c_int = 0x2000;
pub const O_SYMLINK: c_int = 0x4000;
pub const O_NOFOLLOW: c_int = 0x8000;
pub const O_ACCMODE: c_int = O_RDONLY | O_WRONLY | O_RDWR;
