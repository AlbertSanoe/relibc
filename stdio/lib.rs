#[no_mangle]
pub extern "C" fn clearerr(arg1: *mut FILE) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ctermid(arg1: *mut libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn cuserid(arg1: *mut libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fclose(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fdopen(arg1: libc::c_int,
                  arg2: *const libc::c_char) -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn feof(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ferror(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fflush(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fgetc(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fgetpos(arg1: *mut FILE, arg2: *mut fpos_t)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fgets(arg1: *mut libc::c_char,
                 arg2: libc::c_int, arg3: *mut FILE)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fileno(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn flockfile(arg1: *mut FILE) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fopen(arg1: *const libc::c_char,
                 arg2: *const libc::c_char) -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fprintf(arg1: *mut FILE, arg2: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fputc(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fputs(arg1: *const libc::c_char, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fread(arg1: *mut libc::c_void, arg2: usize, arg3: usize,
                 arg4: *mut FILE) -> usize {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn freopen(arg1: *const libc::c_char,
                   arg2: *const libc::c_char, arg3: *mut FILE)
     -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fscanf(arg1: *mut FILE, arg2: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fseek(arg1: *mut FILE, arg2: libc::c_long,
                 arg3: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fseeko(arg1: *mut FILE, arg2: off_t, arg3: libc::c_int)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fsetpos(arg1: *mut FILE, arg2: *const fpos_t)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ftell(arg1: *mut FILE) -> libc::c_long {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ftello(arg1: *mut FILE) -> off_t {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ftrylockfile(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn funlockfile(arg1: *mut FILE) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn fwrite(arg1: *const libc::c_void, arg2: usize,
                  arg3: usize, arg4: *mut FILE) -> usize {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getc(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getchar() -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getc_unlocked(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getchar_unlocked() -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getopt(arg1: libc::c_int,
                  arg2: *const *const libc::c_char,
                  arg3: libc::c_char) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn gets(arg1: *mut libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn getw(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn pclose(arg1: *mut FILE) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn perror(arg1: *const libc::c_char) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn popen(arg1: *const libc::c_char,
                 arg2: *const libc::c_char) -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn printf(arg1: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putc(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putchar(arg1: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putc_unlocked(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putchar_unlocked(arg1: libc::c_int)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn puts(arg1: *const libc::c_char) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn putw(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn remove(arg1: *const libc::c_char)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn rename(arg1: *const libc::c_char,
                  arg2: *const libc::c_char)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn rewind(arg1: *mut FILE) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn scanf(arg1: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn setbuf(arg1: *mut FILE, arg2: *mut libc::c_char) {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn setvbuf(arg1: *mut FILE, arg2: *mut libc::c_char,
                   arg3: libc::c_int, arg4: usize)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn snprintf(arg1: *mut libc::c_char, arg2: usize,
                    arg3: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn sprintf(arg1: *mut libc::c_char,
                   arg2: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn sscanf(arg1: *const libc::c_char,
                  arg2: *const libc::c_char, ...)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn tempnam(arg1: *const libc::c_char,
                   arg2: *const libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn tmpfile() -> *mut FILE {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn tmpnam(arg1: *mut libc::c_char)
     -> *mut libc::c_char {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn ungetc(arg1: libc::c_int, arg2: *mut FILE)
     -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn vfprintf(arg1: *mut FILE, arg2: *const libc::c_char,
                    va_list: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn vprintf(arg1: *const libc::c_char,
                   va_list: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn vsnprintf(arg1: *mut libc::c_char, arg2: usize,
                     arg3: *const libc::c_char,
                     va_list: libc::c_int) -> libc::c_int {
    unimplemented!();
}

#[no_mangle]
pub extern "C" fn vsprintf(arg1: *mut libc::c_char,
                    arg2: *const libc::c_char,
                    va_list: libc::c_int) -> libc::c_int {
    unimplemented!();
}

