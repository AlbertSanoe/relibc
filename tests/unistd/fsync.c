#include <unistd.h>
#include <fcntl.h>
#include <stdio.h>

int main(void) {
    int fd = open(".", 0, 0);
    int status;
    status = fsync(fd);
    printf("fsync exited with status code %d\n", status);
    close(fd);
}
