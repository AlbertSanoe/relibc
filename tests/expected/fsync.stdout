fsync exited with status code 0
